packages/core/src/hierarchy.ts

```diff
@@ -70,4 +72,20 @@ export class Hierarchy {
     }
     return false
   }
+
+  extendsOfClass<T extends Obj>(_class: Ref<Class<T>>): Array<Ref<Class<Obj>>> {
+    return this.extends.get(_class) ?? []
+  }
+
+  private addExtends<T extends Obj>(_class: Ref<Class<T>>): void {
+    const hierarchy = this.getAncestors(_class as Ref<Class<Obj>>)
+    for (const cls of hierarchy) {
+      const list = this.extends.get(cls)
+      if (list === undefined) {
+        this.extends.set(cls, [_class as Ref<Class<Obj>>])
+      } else if (!list.includes(_class)) {
```

Comment:
`list` will never include a class, is not it?

Reply:
Never in normal case only, but if the transaction is applied twice, or the objectId is repeated in different transactions, it will cause duplicate data

Reply:
Create transaction can't happen twice

Reply:
and objectId with create transaction can't appear twice, if you want to preserve this check I would better recommend to throw exception if there is class in the list, but for performance reasons I'd prefer to remove this check